utils/patientmatching/rule_based_patient_matcher.bal

```diff
@@ -0,0 +1,282 @@
+        }
+        sql:ParameterizedQuery queryString = ``;
+        queryString.strings = [qry];
+        stream<record {}, sql:Error?> resultStream = dbClient->query(queryString);
+        return resultStream;
+    }
+
+    isolated function getMPIConfigData(ConfigurationRecord config) returns MPIDbConfig|error {
+        json|error masterPatientIndexHost = config?.masterPatientIndexHost;
+        json|error masterPatientIndexPort = config?.masterPatientIndexPort;
+        json|error masterPatientIndexDb = config?.masterPatientIndexDb;
+        json|error masterPatientIndexDbUser = config?.masterPatientIndexDbUser;
+        json|error masterPatientIndexDbPassword = config?.masterPatientIndexDbPassword;
+        if masterPatientIndexHost is error {
+            return throwFHIRError("Error occurred while getting the masterPatientIndexHost from the config.json file.", cause = masterPatientIndexHost);
+        }
+        if masterPatientIndexPort is error {
+            return throwFHIRError("Error occurred while getting the masterPatientIndexPort from the config.json file.", cause = masterPatientIndexPort);
+        }
+        if masterPatientIndexDb is error {
+            return throwFHIRError("Error occurred while getting the masterPatientIndexDb from the config.json file.",cause = masterPatientIndexDb);
+        }
+        if masterPatientIndexDbUser is error {
+            return throwFHIRError("Error occurred while getting the masterPatientIndexDbUser from the config.json file.",cause = masterPatientIndexDbUser);
+        }
+        if masterPatientIndexDbPassword is error {
+            return throwFHIRError("Error occurred while getting the masterPatientIndexDbPassword from the config.json file.", cause = masterPatientIndexDbPassword);
+        }
+        MPIDbConfig mpiDbConfig = {
+            host: <string>masterPatientIndexHost,
+            port: <int>masterPatientIndexPort,
+            database: <string>masterPatientIndexDb,
+            username: <string>masterPatientIndexDbUser,
+            password: <string>masterPatientIndexDbPassword
+        };
+        return mpiDbConfig;
+    }
+
+    isolated function getSQLQuery(r4:Patient patient, RulesRecord rulesTable, ConfigurationRecord config) returns (error|string) {
+        json|error MPIColumnNames = config?.masterPatientIndexColumnNames;
+        json|error MPITableName = config?.masterPatientIndexTableName;
+        if MPIColumnNames is error {
+            return throwFHIRError("Error occurred while getting the MPIColumnNames from the config.json file.", cause = MPIColumnNames);
+        }
+        if MPITableName is error {
+            return throwFHIRError("Error occurred while getting the MPITableName from the config.json file.", cause = MPITableName);
+        }
+        json[] parameters = <json[]>MPIColumnNames;
+        string tableName = <string>MPITableName;
+        string query = "SELECT * FROM ";
+        query = string `${query} ${tableName} WHERE `;
```

Comment:
```suggestion
        string query = "SELECT * FROM ";
        query = string `SELECT * FROM  ${tableName} WHERE `;
```

Reply:
Fixed in bd514ee7ad5e188fbaa6a4484af40fc238568f8d